### PR TITLE
feat: add shorten url protocol

### PR DESCRIPTION
**Pull request by TimoGlastra**
Adds a protocol to request a shortened variant of an url. It can be used for shortening generic urls, but also has specific goal codes for url shortening that adheres to the shortening standard as described in the OOB v1 and v2 protocols.

### CODE REVIEW

File: features/0746-shorten-url/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+- Authors: [Timo Glastra](mailto:timo@animo.id) (Animo Solutions)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2022-07-26
+- Status Note: Initial version
+- Start Date: 2022-07-20
+- Tags: [feature](/tags.md#feature), [protocol](/tags.md#protocol)
+
+## Summary
+
+A protocol to request a shortened URL for a given URL.
```
Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:12]: Example, an agent requesting a shorted URL from a Mediator.

File: features/0746-shorten-url/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+When the shortened URL is requested, the **url-shortener** MUST respond with a status code of `301` or `302` and include a `Location` header specifcing header specifying the long url.
+
+## Drawbacks
+
+- Using another agent as url shortener leaks some information about the connections an agent makes to the other agent. However in the case of a mediator, where this protocol will probably be most often used, as mobile edge agents can't create shortened urls themeslves, and the mediator already keeps a list of all the registered keys from RFC 0211, this doesn't leak a lot of extra information.
+- Using the mediator as a url shortener adds extra dependency on the mediator to be avaiable and act in your favor. However, as the mediator already plays a crucial role in routing messages, it doesn't add a lot of extra trust in the mediator.
+
+## Prior art
+
+-
```
Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:155]: https://en.wikipedia.org/wiki/URL_shortening

File: features/0746-shorten-url/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+Message sent by the **long-url-provider** to the **url-shortener** to request a shortened version of an url.
+
+```json
+{
+  "@type": "https://didcomm.org/shorten-url/1.0/request-shortened-url",
+  "@id": "<UUID>",
+  "url": "<URL>",
+  "requested_validity_seconds": <INTEGER>,
+  "goal_code": "<GOAL_CODE>"
```
Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:52]: Perhaps add a vanity or beautify string here? For example, I would like to request....

example.org/my-brand-connect

Something like

"vanity": "<URL-SAFE-STRING>"

"vanity": "my-brand-connect"

File: features/0746-shorten-url/README.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+#### Invalidate Shortened URL
+
+Message sent by the **long-url-provider** to the **url-shortener** to invalidate a shortened url. This is useful if the short url is no longer needed and reduces the chance of unwanted exposure. If the **long-url-provider** is authorized to invalidate the shortened url, the **url-shortener** SHOULD invalidate the url due to privacy concerns.
+
+```json
+{
+  "@type": "https://didcomm.org/shorten-url/1.0/invalidate-shortened-url",
+  "@id": "<UUID>",
+  "shortened_url": "<SHORTENED-URL>"
+}
```
Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:118]: Perhaps add an expiration for the url.

Something like, this url can be re-used in 90 minutes or 40 days, or 1 year, or never.

The concept is that even though I don't want the link to redirect to me anymore, I don't want someone else using this link and redirecting to someone else's link.

File: features/0746-shorten-url/README.md (hunk 2 above)
Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:155]: https://github.com/hyperledger/aries-rfcs/tree/main/features/0434-outofband#url-shortening

Comment (reflectivedevelopment) [inline features/0746-shorten-url/README.md:155]: https://github.com/hyperledger/aries-rfcs/blob/145af6f948d5b1a3c152599cf9a41750f66d7e4e/concepts/0268-unified-didcomm-agent-deeplinking/README.md#invitation-page

**Comment by rodolfomiranda** [comment]
Should it also includes message examples in DIDComm v2 format?

**Comment by TimoGlastra** [comment]
> Should it also include message examples in DIDComm v2 format?

I could also submit it as a DIDComm v2 protocol and add DIDComm v1 examples to it? It's not really related to Aries this protocol

**Comment by rodolfomiranda** [comment]
> > Should it also include message examples in DIDComm v2 format?
> 
> I could also submit it as a DIDComm v2 protocol to https://didcomm.org and add DIDComm v1 examples to it? It's not really related to Aries this protocol

Yes, that can be a full description or a link to that one if it has full examples. I wonder if RFC should start covering both types in order to be ready for DIDComm v2 adoption.

File: features/0746-shorten-url/README.md (hunk 3 above)
Comment (TimoGlastra) [reply features/0746-shorten-url/README.md:52]: This seems to be a feature that is available in most url shorteners. Adding it!

**Comment by TimoGlastra** [comment]
Superseded by https://github.com/decentralized-identity/didcomm.org/pull/57

**Pull request closed without merging**